### ISSUE TITLE
Logtrail hostname selection not working with filebeat 

### ISSUE THREAD

**Issue by tssit** [sivasamyk/logtrail#275]
Hi 
ES/Filebeat/kibana/Logtrail Version 6.2.4

Logtrail hostname selection not working with filebeat-index and It works in All Systems selection I checked browser console which shows timestamp null

Browser console debug
![logtrail](https://user-images.githubusercontent.com/29948853/39698321-ff4e47b8-5211-11e8-82b3-5154feded8bb.JPG)

Logtrail.json file

`  "version" : 1,
  "index_patterns" : [
    {
      "es": {
        "default_index": "filebeat-*"
      },
      "tail_interval_in_seconds": 10,
      "es_index_time_offset_in_seconds":0,
"display_timezone": "local",
"display_timestamp_format": "MMM DD HH:mm:ss",
      "max_buckets": 500,
      "default_time_range_in_days" : 0,
      "max_hosts": 100,
      "max_events_to_keep_in_viewer": 5000,
      "fields" : {
        "mapping" : {
            "timestamp" : "@timestamp",
            "hostname" : "host",
            "message": "message"
        },
        "message_format": "{{{message}}}"
      },
      "color_mapping" : {
      }
    },
    {
      "es": {
        "default_index": "logstash-zookeeper"
      },
      "tail_interval_in_seconds": 10,
      "es_index_time_offset_in_seconds": 0,
      "display_timezone": "local",
      "display_timestamp_format": "MMM DD HH:mm:ss",
      "max_buckets": 500,
      "default_time_range_in_days" : 0,
      "max_hosts": 100,
      "max_events_to_keep_in_viewer": 5000,
      "fields" : {
        "mapping" : {
            "timestamp" : "@timestamp",
            "hostname" : "host",
            "program": "logger_name",
            "message": "log_message"
        },
        "message_format": "{{{log_message}}}"
      },
      "color_mapping" : {
      }
    }
  ]
}
`

``

**Comment by Azbests** [comment]
I can confirm this behaviour. I can see hosts in systems list, but when selecting any I get the same message. Only selection that works is "All Systems". 

Only thing I have changed in my logtrail.json is pretty much just default index to same value as yours "filebeat-*"

**Comment by vitalykarasik** [comment]
You should update  logtrail.json  with your actual hostname field. 
I had similar problem and fixed it by updating "hostname" in config.
See https://github.com/sivasamyk/logtrail/blob/master/docs/how_to.md#1-hostname-field-need-to-be-of-type-keyword for more.

Regards,
Vitaly

**Comment by ceeeekay** [comment]
I've been having the same issue for a while - it doesn't matter what the hostname field is set to.

https://github.com/sivasamyk/logtrail/issues/248#issuecomment-374065608

**Comment by tssit** [comment]
I fixed it by changing host name field type to text in template

` "beat": {
            "properties": {
              "hostname": {
                "type": "text",
                "fields": {
                  "keyword": {
                    "type": "keyword",
                    "ignore_above": 256
                  }
                }
              },
              "name": {
                "type": "text",
                "fields": {
                  "keyword": {
                    "type": "keyword",
                    "ignore_above": 256
                  }
                }
              }`

**Comment by ceeeekay** [comment]
That's not really a fix if you want to use the host field in any Kibana aggregations.

**Comment by pureclouds** [comment]
I have the same issue. Think problem is in default filebeat template.

From default **logstash** template:

```
        "string_fields" : {
          "match" : "*",
          "match_mapping_type" : "string",
          "mapping" : {
            "type" : "text", "norms" : false,
            "fields" : {
              "keyword" : { "type": "keyword", "ignore_above": 256 }
            }
          }
        }
```

and from default **filebeat** template:

```
            "strings_as_keyword" : {
              "mapping" : {
                "ignore_above" : 1024,
                "type" : "keyword"
              },
              "match_mapping_type" : "string"
            }

...

          "beat" : {
            "properties" : {
              "timezone" : {
                "type" : "keyword",
                "ignore_above" : 1024
              },
              "version" : {
                "type" : "keyword",
                "ignore_above" : 1024
              },
              "name" : {
                "type" : "keyword",
                "ignore_above" : 1024
              },
              "hostname" : {
                "ignore_above" : 1024,
                "type" : "keyword"
              }
            }
          },

```

@sivasamyk What you think about ?

**Comment by sivasamyk** [comment]
Made some changes in https://github.com/sivasamyk/logtrail/releases/tag/v0.1.28 release to handle keyword fields in logtrail. Can you verify with this release. Also https://github.com/sivasamyk/logtrail/blob/master/logtrail.json this config works for me with filebeat.

**Comment by ceeeekay** [comment]
@sivasamyk Just tested 6.3.0-0.1.28 and host selection is still not working for me.  I don't have a `host.keyword` field, just a `host` field which is mapped as a keyword. 

This update has also broken my ability to click a program as it's now adding .keyword to the end of the query. There are no .keyword fields in my index.  Wouldn't it be better to have the user specify .keyword in their config if that's what they have? Not everyone is using the default mappings, as it wastes a lot of disk space.

tl;dr: This update has made things worse for me. I just want to be able to select `host` and `program`.

Logtrail field config and `host` mapping:
```
        "mapping" : {
            "timestamp" : "@timestamp",
            "hostname" : "host",
            "program": "program",
            "message": "syslog_message"
        },
```
```
{
  "sysadmins-2018.07.02" : {
    "mappings" : {
      "doc" : {
        "host" : {
          "full_name" : "host",
          "mapping" : {
            "host" : {
              "type" : "keyword"
            }
          }
        }
      }
    }
  }
}
```

**Comment by sivasamyk** [comment]
@ceeeekay Can you add `keyword_prefix: ""` to your logtrail.json? Not sure if you have specified this in logtrail.json. Setting it to empty string will stop logtrail from appending `.keyword` to host and program field names.

**Comment by sivasamyk** [comment]
By default `keyword_suffix` is set to `keyword` here https://github.com/sivasamyk/logtrail/blob/ace92920d3a4b1abfaccfbab143b7594f4ba2b04/logtrail.json#L24.  Change this to `keyword_suffix:""`

**Comment by ceeeekay** [comment]
@sivasamyk Yes, I realised that slightly too late, and it's solved all my problems. Puppet had copied my old config over the top of the new install.

Everything works great. Thanks very much!
